3-evaluator.md

```diff
@@ -0,0 +1,144 @@
+
+# Evaluators
+
+## Synopsis
+
+Provide an `Evaluators` constructor that produces a new `eval` function,
+`Function` constructor, and `Module` constructor such that execution contexts
+generated from these evaluators refer back to this set of evaluators, with a
+given global object and virtualized host behavior for dynamic import in script
+contexts.
+
+## Interfaces
+
+```ts
+interface Evaluators {
+  constructor({
+    globalThis?: Object,
```

Comment:
This might be problematic. In the past, implementers agreed to allow the creation of global object as ordinary object (which is the case of ShadowRealm today), but providing a global object from user-land, I don't think that will be possible. What if that object is a proxy? There are certain qualities of a global object (specially if it is associated to a module instance) that implementers will like to preserve, and letting them creating such object (ordinary or not) seems important. This is not a deal-breaker, but will require some adjustment of this API.

Reply:
Agreed. I figure we should ask for this and change it only if there are strong objections, since this is much easier for users to understand than the alternative (which would need separate argument patterns for shared globals and fresh globals).